src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs

```diff
@@ -79,7 +83,7 @@ internal static OtlpLogs.LogRecord ToOtlpLog(this LogRecord logRecord, SdkLimitO
 
             var attributeValueLengthLimit = sdkLimitOptions.AttributeValueLengthLimit;
             var attributeCountLimit = sdkLimitOptions.AttributeCountLimit ?? int.MaxValue;
-
+#if EXPOSE_EXPERIMENTAL_FEATURES
```

Comment:
Now that we are marking it experimental, could we rename the EventId related attributes to `dotnet.ilogger.event_id` and `dotnet.ilogger.event_name` avoid having users run into these issues #4404 ?

We should be able to rename their names again in future since these would only be exported in pre-release versions of the exporter.

Reply:
I'd be ok with that, but I'm looking to isolate this  PR from any conversation of changing attribute names.

Seemed folks felt we need to more deeply consider the name and maybe even consider using the event.domain/event.name conventions. I think continuing that conversation on a different PR helps us focus on the conversations that are actually required for our next stable release.

Reply:
Got it. I also wanted to talk about using the compiler constant to change the behavior of the code.

If I remember correctly, we had discussed in the SIG meeting that the changes introduced in #4735 are only meant for offering experimental features that require public APIs to be added. For internal behavioral changes, we would want to use environment variables.

Personally, I'm okay with using the newly added compiler constant even for internal behavioral changes such as these. This does prevent us from allowing a user to use a stable version of the package with the experimental internal behavior configuration. I think we should also work on formulating a guideline for ourselves when adding experimental features.

Reply:
To me it's less about creating guidelines for using a pattern only for certain scenarios. My priority is for us to get better at shipping code more frequently. Awhile back I mentioned that I'd love for us to get to a point where we could ship a stable release on a regular cadence, e.g., monthly. To get there, we need to form different habits and practices. Most importantly I think the practice of _always_ leaving the main branch in a shippable state becomes really important.

We now have a number of tools at our disposal:

1. Separating functionality into their own pre-release versioned packages.
2. `OTEL_DOTNET_EXPERIMENTAL_*` feature flags
3. The `EXPOSE_EXPERIMENTAL_FEATURES` compiler directive

As all of this applies to this PR, I think the tool of the compiler directive is an effective means to move the OTLP exporter into a shippable state allowing us to consider more deeply the possibility of feature flagging this functionality later. I think feature flagging might be a good strategy for gating this functionality, but I'm not yet convinced. For example, if these attributes really are ILogger specific, then I actually think it would be ideal if we could find a way to ship this functionality in a separate package, e.g., OpenTelemetry.Instrumentation.ILogger.

Though, back to your original point about creating guidelines.. I do suspect we will find that certain patterns are best for certain scenarios, but I think staying flexible keeping the priority on shipping code is most important.

Reply:
In general, compiler directive way should not be used to control internal behavior changes IMO. It would lead to unexpected behavior during runtime after users upgrade/auto-upgrade.

Reply:
I'd still be open to ripping this code out entirely if anyone feels really strongly about that.